apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -0,0 +1,65 @@
+import { TRPCError } from "@trpc/server"
+
+import {
+  getChildrenSchema,
+  getMetadataSchema,
+  moveSchema,
+} from "~/schemas/resource"
+import { protectedProcedure, router } from "~/server/trpc"
+import { db } from "../database"
+
+export const resourceRouter = router({
+  getMetadataById: protectedProcedure
+    .input(getMetadataSchema)
+    .query(async ({ input: { resourceId } }) => {
+      const resource = await db
+        .selectFrom("Resource")
+        .where("Resource.id", "=", String(resourceId))
+        .select(["permalink", "Resource.id", "Resource.title"])
+        .executeTakeFirst()
+
+      if (!resource) {
+        throw new TRPCError({ code: "NOT_FOUND" })
+      }
+
+      return resource
+    }),
+
+  getChildrenOf: protectedProcedure
+    .input(getChildrenSchema)
+    .query(async ({ input: { resourceId } }) => {
+      let query = db
+        .selectFrom("Resource")
+        .select(["title", "permalink", "type", "id"])
+        .orderBy("type", "asc")
+        .orderBy("title", "asc")
+
+      if (resourceId === null) {
+        query = query.where("parentId", "is", null)
+      } else {
+        query = query.where("Resource.parentId", "=", String(resourceId))
+      }
+
+      return query.execute()
+    }),
+  move: protectedProcedure
+    .input(moveSchema)
+    .mutation(async ({ input: { movedResourceId, destinationResourceId } }) => {
+      await db
+        .updateTable("Resource")
+        .where("id", "=", String(movedResourceId))
+        .set({ parentId: String(destinationResourceId) })
+        .execute()
+      return db
+        .selectFrom("Resource")
+        .where("id", "=", String(movedResourceId))
+        .select([
+          "parentId",
+          "Resource.id",
+          "Resource.type",
+          "Resource.permalink",
+          "Resource.title",
+        ])
+        .executeTakeFirst()
```

Comment:
should this be in a transaction? read after write consistency guarantee. idk if can actually be affected or not but no harm just wrapping in tx haha